backend/apid/graphql/entity.go

```diff
@@ -35,7 +35,8 @@ func (*entityImpl) ID(p graphql.ResolveParams) (string, error) {
 // Metadata implements response to request for 'metadata' field.
 func (*entityImpl) Metadata(p graphql.ResolveParams) (interface{}, error) {
```

Comment:
I'm sure there is a reason, but wouldn't it be safer to have `Metadata()` return `(v2.ObjectMeta, error)` instead? Then this would have become a compile time error.

Reply:
Absolutely agree, in theory these are the sort of mistakes using a strongly typed language should help us avoid entirely. In general I'd always prefer using a concrete type.

However, this is current a deficiency in the code generator package (`graphql/generator`). The _field resolver_ interfaces are generated from the GraphQL IDL definitions found in `api/graphql/schema`, (specifically in this case [schema/entity.graphl](https://github.com/sensu/sensu-go/blob/c5a0ffe/backend/apid/graphql/schema/entity.graphql#L12-L13).) From the generator's perspective, the problem is that there no hints given about which (if any) Golang type the GraphQL type represents, as such when it encounters the `ObjectMeta` GraphQL type it has no way of knowing whether it is referring to the `ObjectMeta` type in `core/v2`, or the `ObjectMeta` type in the Kubernetes package, or whatever other package may export a type by the same name.

The obvious solution would be to implement a typemap or some other mechanism so that the generator can lookup what Golang type the GraphQL type represents. Then we can generate interfaces that are more strict and help us avoid what should be easily avoidable bugs.